core/templates/dev/head/domain/utilities/UrlInterpolationService.ts

```diff
@@ -167,8 +167,9 @@ angular.module('oppia').factory('UrlInterpolationService', [
        * Given an story id returns the complete url path to that image.
        */
       getStoryUrl: function(storyId) {
-        validateResourcePath(storyId);
-        return '/story' + storyId;
+        var returnUrl = '/story/' + storyId;
```

Comment:
Consider instead calling interpolateUrl() here, with the appropriate URL template. I think it encodes URI components.

Reply:
Done, deleted this function and created a new URL template that can be used with interpolateUrl anywhere,